inception-workload-dynamic/src/main/java/de/tudarmstadt/ukp/inception/workload/dynamic/page/MonitoringPage.java

```diff
@@ -0,0 +1,524 @@
+        } else {
+            currentProject = null;
+        }
+
+
+        //Get the current user
+        User user = userRepository.getCurrentUser();
+        //Check if Project exists
+        if (currentProject != null)
+        {
+
+
+
+
+            //Check if user is allowed to see the monitoring page of this project
+            if (currentProject != null &&
+                !(projectService.isCurator(currentProject.getObject(), user)
+                || projectService.isManager(currentProject.getObject(), user)))
+            {
+                //Required even in case of error due to wicket
+                workload.add(new ModalWindow("modalWindow"));
+                workload.add(new NumberTextField("defaultDocumentsNumberTextField"));
+                workload.add(new EmptyPanel("dataTable"));
+                error("You have no permission to access project [" + currentProject.getObject().getId() + "]");
+                setResponsePage(getApplication().getHomePage());
+
+            } else {
+                initialize();
+            }
+
+        } else {
+            //Project does not exists, returning to homepage
+            error("Project [" + projectParameter + "] does not exist");
+            setResponsePage(getApplication().getHomePage());
+
+        }
+
+    }
+
+    public void initialize()
+    {
+
+        //Header of the page
+        Label name = new Label("name",currentProject.getObject().getName());
+        add(name);
+
+
+        //Headers of the table
+        List<String> headers = new ArrayList<>();
+        headers.add(getString("Document"));
+        headers.add(getString("Finished"));
+        headers.add(getString("Processed"));
+        headers.add(getString("Metadata"));
+
+        //Data Provider for the table
+        DataProvider dataProvider = new DataProvider(documentService.
+            listSourceDocuments(currentProject.getObject()),
+            headers, documentService.listAnnotationDocuments
+            (currentProject.getObject()));
+
+        //Init defaultDocumentsNumberTextField
+        NumberTextField defaultNumberDocumentsTextField =
+            new NumberTextField("defaultDocumentsNumberTextField", Integer.class);
+
+        //Set minimum value for input
+        defaultNumberDocumentsTextField.setMinimum(1);
+        //After upstream change
+        defaultNumberDocumentsTextField.setRequired(true);
+
+        ModalWindow modalWindow = new ModalWindow("modalWindow");
+        workload.add(modalWindow);
+
+
+
+
+        //Get value for the project and set it accordingly
+        //TODO get correct value after upstream change
+        defaultNumberDocumentsTextField.setDefaultModel(new CompoundPropertyModel<Integer>(6));
+        defaultNumberDocumentsTextField.setConvertEmptyInputStringToNull(false);
+
+        //add AJAX event handler on changing input value
+        defaultNumberDocumentsTextField.add(new OnChangeAjaxBehavior()
```

Comment:
I think this can be done using our `LambdaAjaxFormComponentUpdatingBehavior`.

Reply:
Parsing a string and setting the model value should not be necessary.
If you use the component in a form with a submit button, then saving the value into the model (and converting it) is done by Wicket automatically.
If you use the component outside a form, then just adding the `LambdaAjaxFormComponentUpdatingBehavior` should do the trick. e.g.:

```
            // Just update the selection
            repositoryCombo.add(new LambdaAjaxFormComponentUpdatingBehavior("change"));
```